core/src/test/java/io/questdb/griffin/engine/functions/str/SubStringFunctionFactoryTest.java

```diff
@@ -0,0 +1,117 @@
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.str;
+
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.engine.AbstractFunctionFactoryTest;
+import io.questdb.std.Numbers;
+import org.junit.Test;
+
+import static org.junit.Assert.fail;
+
+public class SubStringFunctionFactoryTest extends AbstractFunctionFactoryTest {
+
+    @Override
+    protected FunctionFactory getFunctionFactory() {
+        return new SubStringFunctionFactory();
+    }
+
+    @Test
+    public void testSimple() throws Exception {
+        assertQuery(
+                "k\tsubstring\tlength\n" +
+                        "JWCPSWHYRXPEHNRXGZSXUXIB\tNRXGZS\t6\n" +
+                        "GPGWFFYUDE\t\t0\n" +
+                        "QEHBHFOWLPDXYSBEOUOJSH\tSBEOUO\t6\n" +
+                        "EDRQQULOFJGETJRSZSRY\tJRSZSR\t6\n" +
+                        "BVTMHGOOZZVDZJMYICCXZ\tJMYICC\t6\n" +
+                        "ICWEKGHVUVSDOTSEDYYCTGQ\tTSEDYY\t6\n" +
+                        "YXWCKYLSUWDSWUGSH\tUGSH\t4\n" +
+                        "NVTIQBZXIOVIKJSMSSUQSR\tJSMSSU\t6\n" +
+                        "\t\t-1\n" +  // null
+                        "VVSJOJIPHZEPIHVLTO\tHVLTO\t5\n" +
+                        "JUMLGLHMLLEOY\t\t0\n" +
+                        "\t\t-1\n" +  // null
+                        "IPZIMNZZRMFMBEZG\tEZG\t3\n" +
+                        "VDKFLOPJOXPKRGIIHY\tGIIHY\t5\n" +
+                        "OQMYSSMPGLUOHNZHZS\tNZHZS\t5\n",
+                "select k, substring(k,14,6), length(substring(k,14,6)) from x",
+                "create table x as (select rnd_str(10,25,1) k from long_sequence(15))",
+                null,
+                true,
+                true,
+                true
+        );
+    }
+
+    @Test
+    public void testNullOrEmptyStr() throws Exception {
+        call(null, 2, 4).andAssert(null);
+        call("", 2, 4).andAssert("");
+    }
+
+    @Test
+    public void testZeroOrInvalidLength() throws Exception {
+        call("foo", 3, 0).andAssert("");
+        call(null, 3, 0).andAssert(null);
+        call("foo", 3, Numbers.INT_NaN).andAssert(null);
+
+        try {
+            call("foo", 3, -1).andAssert(null);
+            fail("non-const negative len is not allowed");
+        } catch (RuntimeException e) {
+            // negative substring length is not allowed
+        }
+
+        try {
+            assertQuery(
+                    null,
+                    "select substring('foo',1,-6)",
+                    null,
+                    true
+            );
+            fail("const negative len is not allowed");
+        } catch (SqlException e) {
+            // negative substring length is not allowed
+        }
+    }
+
+    @Test
+    public void testNonPositiveStart() throws Exception {
```

Comment:
How about call("foo", -3, 3) ?

Reply:
That returns an empty string, same as Postgres